Subject: [PATCH] Fix escaping of arguments passed to LaunchProcess() in JavaBinaryLauncher::CreateClasspathJar on Windows
---
Index: src/tools/launcher/java_launcher.cc
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/tools/launcher/java_launcher.cc b/src/tools/launcher/java_launcher.cc
--- a/src/tools/launcher/java_launcher.cc	(revision 33650c6ba5589643503ae240aa67e6527713f33e)
+++ b/src/tools/launcher/java_launcher.cc	(date 1753388457972)
@@ -280,8 +280,8 @@
   wstring jar_bin = this->Rlocation(this->GetLaunchInfoByKey(JAR_BIN_PATH));
   vector<wstring> arguments;
   arguments.push_back(L"cvfm");
-  arguments.push_back(manifest_jar_path);
-  arguments.push_back(jar_manifest_file_path);
+  arguments.push_back(bazel::windows::WindowsEscapeArg(manifest_jar_path));
+  arguments.push_back(bazel::windows::WindowsEscapeArg(jar_manifest_file_path));
 
   if (this->LaunchProcess(jar_bin, arguments, /* suppressOutput */ true) != 0) {
     die(L"Couldn't create classpath jar: %s", manifest_jar_path.c_str());
