From 4e2d58804b4e15a04787eed3879d6049ff03e8ed Mon Sep 17 00:00:00 2001
From: Ilia K <ki.stfu@gmail.com>
Date: Fri, 29 Aug 2025 17:38:48 +0200
Subject: [PATCH] Fix compilation in third_party/zlib/zutil.h

---
 third_party/zlib/zutil.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/third_party/zlib/zutil.h b/third_party/zlib/zutil.h
index 902a304cc2..c358516c13 100644
--- a/third_party/zlib/zutil.h
+++ b/third_party/zlib/zutil.h
@@ -144,7 +144,6 @@ extern z_const char * const z_errmsg[10]; /* indexed by 2-zlib_error */
 #      include <unix.h> /* for fdopen */
 #    else
 #      ifndef fdopen
-#        define fdopen(fd,mode) NULL /* No fdopen() */
 #      endif
 #    endif
 #  endif
-- 
2.50.1

